### PR TITLE
[qa] Increase wallet-dump RPC timeout

### PR DESCRIPTION
**Pull request by ryanofsky**
Increase wallet-dump RPC timeout from 30 seconds to 15 minutes. 15 minutes was
chosen because it's the same timeout used by the pruning test. This avoids a
timeout error that seemed to happen regularly (around 50% of builds) on a
particular jenkins server during the first getnewaddress RPC call made by the
test.

### CODE REVIEW

**Comment by sdaftuar** [comment]
(FYI this is my jenkins instance that has been seeing this test fail repeatedly over the last couple months.)

For some additional color, I timed the RPC calls in wallet-dump.py on a single run on the machine in question -- it happened to pass -- and the first getnewaddress() call took 20 seconds, and the encrypt wallet call took just over 29 seconds.  So the 30 second default timeout is cutting it pretty close.

**Comment by laanwj** [comment]
Concept ACK

File: qa/rpc-tests/wallet-dump.py [hunk 1]
```diff
@@ -61,7 +61,11 @@ def __init__(self):
         self.extra_args = [["-keypool=90"]]
 
     def setup_network(self, split=False):
-        self.nodes = start_nodes(self.num_nodes, self.options.tmpdir, self.extra_args)
+        # Use long, 15-minute timeout because the initial getnewaddress RPC can
```
Comment (MarcoFalke) [inline qa/rpc-tests/wallet-dump.py]: The whole wallet dump test should run in less than 15 seconds. I don't see why a 15 *minute* timeout is necessary.

From travis (confirmed locally):
```
wallet-dump.py                 | True   | 12 s
```

**Comment by MarcoFalke** [comment]
> getnewaddress() call took 20 seconds

How large is the keypool set on the jenkins instance?

**Comment by MarcoFalke** [comment]
utACK for just 82199c4

**Comment by ryanofsky** [comment]
Changed timeout from 15 minutes to one minute, as suggested by @MarcoFalke in IRC.

**Comment by sdaftuar** [comment]
@MarcoFalke Any interest in backporting this to 0.13?  It's minor, but my testing infrastructure would benefit at least.

**Pull request merged**